cdc-accelerator/feature/site/__tests__/smsTemplates.test.js

```diff
@@ -116,4 +114,161 @@ describe('Sms templates test suite', () => {
             }
         }
     })
+    describe('Build test suite', () => {
+        test('SMS templates are built successfully', () => {
+            const mockFs = {
+                [path.join(srcSiteDirectory, 'SmsTemplates', 'otp')]: {
+                    globalTemplates: ['en-default.txt', 'nl.txt'],
+                    templatesPerCountryCode: {
+                        244: ['pt.txt'],
+                    },
+                },
+                [path.join(srcSiteDirectory, 'SmsTemplates', 'tfa')]: {
+                    globalTemplates: ['en-default.txt', 'nl.txt'],
+                },
+            }
+
+            fs.readdirSync.mockImplementation((dirPath) => {
+                const entries = mockFs[dirPath] || {}
+                return Object.keys(entries).concat(entries instanceof Array ? entries : [])
+            })
+
+            fs.existsSync.mockImplementation((path) => !!mockFs[path])
+
+            fs.statSync.mockImplementation((path) => ({
+                isFile: () => typeof mockFs[path] === 'undefined',
+                isDirectory: () => typeof mockFs[path] !== 'undefined',
+            }))
+
+            fs.readFileSync.mockReturnValue('Template content')
+            const writeFileSyncMock = jest.spyOn(fs, 'writeFileSync')
+
+            const smsTemplates = new SmsTemplates(credentials)
+            smsTemplates.build(BUILD_DIRECTORY)
+        })
+
+        test('No SMS templates are built when no files are present', () => {
+            fs.readdirSync.mockReturnValue([])
+            fs.existsSync.mockReturnValue(true)
+            const writeFileSyncMock = jest.spyOn(fs, 'writeFileSync').mockImplementation(() => {})
+
+            const smsTemplates = new SmsTemplates(credentials)
+            smsTemplates.build(BUILD_DIRECTORY)
+
+            expect(writeFileSyncMock).not.toHaveBeenCalled()
+        })
+    })
+    describe('Deploy test suite', () => {
+        test('SMS templates deploy fails with multiple default files in a folder', async () => {
```

Comment:
What is the difference between this test and the one below called 'SMS duplicate Default Files Error'?